### PR TITLE
log: support maxBackups in config.toml

### PR DESCRIPTION
**Pull request by zlacfzy**
### Description

Add maxBackups in config.toml.  
MaxBackups means the log file that would be retained, by default it would be 0, which means there is no limitation.
If set MaxBackups = 240, only logs of the most recent 240 hours(10 days) would be kept, since the default rotate time is 1 hour.
See more: #2165.

### Rationale
Add a MaxBackups in Node.LogConfig. 
```
[Node.LogConfig]
FileRoot = ""
FilePath = "bsc.log"
MaxBytesSize = 10485760
Level = "info"
```

### Example
```
[Node.LogConfig]
FileRoot = ""
FilePath = "bsc.log"
MaxBytesSize = 10485760
Level = "info"
MaxBackups = 240
```

### Changes

Notable changes: 
* log: add maxBackups in config.toml

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -118,7 +118,12 @@ func New(conf *Config) (*Node, error) {
 				rotateHours = *conf.LogConfig.RotateHours
 			}
 
-			log.Root().SetHandler(log.NewFileLvlHandler(logFilePath, *conf.LogConfig.MaxBytesSize, *conf.LogConfig.Level, rotateHours))
+			maxBackups := uint(0)
```
Comment (buddh0) [inline node/node.go:121]: fistly
1. maxBackups means number of backups, not include bsc.log self
2. unlike rotateHours, 0 here is a meaningful value, so default it to -1, and check conf.LogConfig.MaxBackups>=0

**Review by buddh0 — APPROVED** [review]

**Comment by zzzckck** [comment]
I would prefer: remove the specified 1 log file on `rotateFile()`
like:
```
func (w *AsyncFileWriter) rotateFile() {
	select {
	case <-w.timeTicker.C:
		if err := w.flushAndClose(); err != nil {
			fmt.Fprintf(os.Stderr, "flush and close file error. err=%s", err)
		}
		if err := w.initLogFile(); err != nil {
			fmt.Fprintf(os.Stderr, "init log file error. err=%s", err)
		}
                // start from here
                if maxBackups > 0 {
		   if err := w.removeExpiredFile(); err != nil {
			fmt.Fprintf(os.Stderr, "remove expired file error. err=%s", err)
		   }
                }
                // end
	default:
	}
}
```

File: log/async_file_writer.go [hunk 2]
```diff
@@ -222,5 +229,25 @@ func (w *AsyncFileWriter) flushAndClose() error {
 }
 
 func (w *AsyncFileWriter) timeFilePath(filePath string) string {
-	return filePath + "." + time.Now().Format("2006-01-02_15")
+	return filePath + "." + time.Now().Format(backupTimeFormat)
+}
+
+func (w *AsyncFileWriter) oldFile(filePath string, maxBackups int) string {
+	return filePath + "." + time.Now().Add(-time.Hour*time.Duration(maxBackups)).Format(backupTimeFormat)
```
Comment (zzzckck) [inline log/async_file_writer.go:236]: RotateHours*time.Duration(maxBackups))

File: log/async_file_writer.go [hunk 3]
```diff
@@ -111,6 +116,8 @@ func (w *AsyncFileWriter) initLogFile() error {
 		return err
 	}
 
+	_ = w.clearBackups()
```
Comment (zzzckck) [inline log/async_file_writer.go:119]: better to check and return error

File: log/async_file_writer.go [hunk 4]
```diff
@@ -111,6 +116,8 @@ func (w *AsyncFileWriter) initLogFile() error {
...
+
 	return nil
 }
```
Comment (zzzckck) [inline log/async_file_writer.go:123]: if `w.initLogFile()` failed, print error log:
fmt.Fprintf(os.Stderr, "init log file error. err=%s", err)

**Review by zzzckck — APPROVED** [review]

**Review by GalaIO — APPROVED** [review]

**Pull request merged**